### ISSUE TITLE
Align with .NET 8.0 SDK analyzers

### ISSUE THREAD

**Issue by idg10** [reaqtive/reaqtor#143]
We're temporarily disabling a couple of diagnostics to ensure the project continues to build. (For some reason, the use of `<AnalysisLevel>7.0</AnalysisLevel>` isn't quite enough to prevent new messages when building on the .NET 8.0 SDK.)

However, our intention is to build cleanly without such workarounds on the latest SDK, so this needs to be addressed properly. So in addition to updating that `AnalysisLevel` build variable from `7.0` to `8.0`, we also need to remove the `<NoWarn>` element (and associated comment) that follows it.

**Comment by HowardvanRooijen** [comment]
This issue predates the .NET 10 migration (PR #155): it was filed against the .NET 8.0 SDK, and the repo now builds on the .NET 10 SDK with `AnalysisLevel` still pinned to `7.0`. The proposal below updates the target from `8.0` to `latest` and folds in the second NoWarn list (`CA2000;CA2213`) that the migration added alongside the original one.
## Proposed execution plan: raise `AnalysisLevel` to `latest` and retire the migration-era NoWarn lists

This is the follow-up promised in `Directory.Build.props` and MIGRATION-NET10.md ("Compromises & deferred work", item 2). It follows the same protocol as the #160 style-enforcement work: measure first, land as a reviewable commit series, keep Debug **and** Release builds at 0 warnings under `TreatWarningsAsErrors=true` throughout, full test suite green after every step.

### Current state (at `feature/dotnet-10`)

- `AnalysisLevel` is pinned to `7.0` (`Directory.Build.props`) while the SDK is .NET 10.
- Two migration-era `NoWarn` lists exist:
  - `IDE0079;IDE0090;CA1305;CA1822;CA1854` (this issue's list)
  - `CA2000;CA2213` (the .NET 10 IDisposable analyzer wave — documented in the props as overwhelmingly false-positive over reactive/operator ownership-transfer patterns; the one genuine leak it found was fixed in code during the migration)

### Step 0 — Measure (numbers below are estimates until this runs)

```
dotnet build All.slnx -c Debug -t:Rebuild -p:AnalysisLevel=latest -p:TreatWarningsAsErrors=false -v q
```

Capture the deduped per-rule histogram; it is the worklist and sizes every step below. Repeat for Release (some rules are asymmetric between configurations — we hit this repeatedly during #160).

### Step 1 — Retire the `IDE0090;CA1305;CA1822;CA1854` wave, one rule per commit

- **IDE0090** (target-typed `new`) — automatable via `dotnet format style`; must respect the #138 exclusions (Optimizers project, USE_SLIM linked files) using the project-scoped technique documented in `eng/Run-DotnetFormat.ps1`.
- **CA1854** (prefer `TryGetValue`) — analyzer code-fix; behavioral no-op but each site deserves a look (double-lookup patterns sometimes hide intentional add-or-update logic).
- **CA1822** (mark members static) — code-fix driven, but **restricted to private/internal members**: making a public member static is a binary-breaking change to the shipped API surface. Public hits get per-site suppressions or are deferred.
- **CA1305** (IFormatProvider) — likely the largest wave. The ~624 existing `string.Format(CultureInfo.InvariantCulture, ...)` sites are already compliant; the wave is the uncultured remainder (`ToString()`, `Parse`, interpolation into culture-sensitive APIs). Mechanical but high-volume; expect `.tt` template sync (121 live templates) for any hits in generated code.
- **IDE0079** (unnecessary suppression) — heads-up: this rule **does not execute in CLI builds** (verified empirically during #160 with a planted-pragma test; SARIF and binary rule inventories both gave false answers). Keeping it in NoWarn is cosmetic for CI; actually acting on it requires an IDE pass. Propose: remove it from NoWarn (harmless), treat cleanup as best-effort in the IDE.

### Step 2 — Triage the `CA2000;CA2213` disposal wave

Re-run the measurement with only these two enabled. Expected outcome given the documented false-positive pattern (operators handing subscriptions/disposables to compositional owners): keep them suppressed **scoped to the operator projects** via `.editorconfig` sections rather than repo-wide NoWarn, with genuine sites fixed or pragma'd individually. If the false-positive density is as bad as the migration notes suggest, an explicit "won't enable, here's why" note in the props replaces the current "deferred" framing.

### Step 3 — Raise `AnalysisLevel` to `latest`

Flip `Directory.Build.props` once steps 1-2 leave the build clean at `latest` in trial mode. This also picks up the throw-helper rules (CA1510-CA1513), which gate step 4.

### Step 4 — Throw-helper modernization (the big mechanical sweep)

Approximate counts from a survey of the live tree (2026-07-03):

| Pattern | Sites | Replacement |
|---|---|---|
| `throw new ArgumentNullException` guard clauses | ~3,172 | `ArgumentNullException.ThrowIfNull` (CA1510) |
| `throw new ArgumentOutOfRangeException` | ~247 | `ArgumentOutOfRangeException.ThrowIfNegative`/`ThrowIfGreaterThan`/… where the guard shape matches |
| `throw new ObjectDisposedException` | ~43 | `ObjectDisposedException.ThrowIf` |
| `throw new ArgumentException` (null-or-empty subset) | ~114 total | `ArgumentException.ThrowIfNullOrEmpty` where applicable |

Execution notes:
- Analyzer code-fixes automate the bulk; the residue (multi-condition guards, custom messages) is manual.
- **T4 lockstep**: any hit inside `*.Generated.cs` must be fixed in the emitting `.tt` too, verified by regeneration diff (`dotnet-t4`, byte-identical modulo BOM) — the technique used throughout the .NET 10 migration.
- Behavioral note: `ThrowIfNull` throws with the *expression* as parameter name via `CallerArgumentExpression` — identical to `nameof(x)` at guard-clause sites, so exception contracts (tested extensively by the ~8,850-test suite) are preserved.

### Step 5 (optional companion, separate issue?)

The MSTest analyzer list (`MSTEST0017;MSTEST0032;MSTEST0036;MSTEST0055`) is deliberately excluded here — it's test-code modernization with different review characteristics (hundreds of test files, zero shipped-code impact).

### Verification bar (every step)

```
dotnet build All.slnx -c Debug -t:Rebuild ; dotnet build All.slnx -c Release   # 0 warnings both
./eng/Run-DotnetFormat.ps1 -Verify -NoRestore                                  # format gate green
dotnet test All.slnx -- --filter "TestCategory!~NonDeterministic_Strong"       # full suite green
```

Sequencing vs PR #155: this is proposed as a **follow-up after #155 merges** (per discussion) — the reformat-scale churn in steps 1 and 4 would otherwise dominate that PR's diff.

**Comment by HowardvanRooijen** [comment]
Executed on `feature/dotnet-10` (PR #155) per the sign-off, as five commits:

- **d7cdb28** — restore assembly-info generation in Rxcel (pre-flight fix found while measuring)
- **2d92e45** — step 1: retire `IDE0079;IDE0090;CA1305;CA1822;CA1854`, all sites fixed (47 IDE0090 via `dotnet format`, the rest by hand; public-API CA1822 suppressed with rationale)
- **2130083** — the small waves that light up at `latest` (CA2022 short-read fixes in the checkpoint serializer — genuine latent bugs; CA1868/1864/1860/1865/1861/2263; CA1515; SYSLIB0050/0056 mirror-API suppressions)
- **951f8bb** — the `ArgumentNullException.ThrowIfNull` sweep: 2,928 sites in 329 files + 103 sites in 23 T4 templates (regeneration-verified byte-identical; two templates found already out of sync from an earlier format pass and fixed). Net −3,790 lines. One behavioral edge case caught by the suite: verbatim identifiers (`@this`) change `CallerArgumentExpression`'s ParamName — those sites pass an explicit `nameof`.
- **211871e** — `AnalysisLevel` → `latest`; `CA2000;CA2213` NoWarn removed **with zero churn** (the disposal wave no longer fires at `latest` — the analyzer false-positive patterns are fixed in the .NET 10 SDK); CA1512/CA1513 conversions (with rationale pragmas where `ObjectName` carries diagnostic identity); `CA1859` deferred as an IDE suggestion with rationale.

Notes for posterity:
- `dotnet format analyzers` applies none of the CA fixes ("Formatted 0 files") — the SDK NetAnalyzers code-fix providers aren't discoverable to it; a purpose-built Roslyn transform + hand-finishing was required.
- Test/perf/playground projects now generate (and discard) XML docs so IDE0005 runs on build — it immediately found 9 dead usings.

Verification bar met at every commit: Debug + Release rebuilds at 0 warnings under full strictness, `dotnet format` gate green, 8,854 tests / 0 failed.
